src/components/loader/loader.component.tsx

```diff
@@ -43,21 +51,13 @@ export const Loader = ({
         l.loader.loading()
       ) : (
         <>
-          <StyledLoaderSquare
-            isInsideButton={isInsideButton}
-            isActive={isActive}
-            size={size}
-          />
-          <StyledLoaderSquare
-            isInsideButton={isInsideButton}
-            isActive={isActive}
-            size={size}
-          />
-          <StyledLoaderSquare
-            isInsideButton={isInsideButton}
-            isActive={isActive}
-            size={size}
-          />
+          {[0, 1, 2].map((index) => (
```

Comment:
Suggestion (non-blocking): you could slightly tweak this so you're only passing a single prop to the `StyledLoaderSquare`s instead of `color` and `loaderNumber` etc

```
const StyledLoaderSquare = styled.div<StyledLoaderSquareProps>`
  ${({ size, isInsideButton, isActive, theme, backgroundColor = "var(--colorsActionMajor500)" }) => css`
...
    background-color: ${backgroundColor};
`};

// in component file
...
const isGradient = varient === "gradient";
...

 <StyledLoaderSquare
            isInsideButton={isInsideButton}
            isActive={isActive}
            size={size}
           backgroundColor={isGradient ? "#13A038" : undefined}
          />
          <StyledLoaderSquare
            isInsideButton={isInsideButton}
            isActive={isActive}
            size={size}
           backgroundColor={isGradient ? "#0092DB" : undefined} 
          />
          <StyledLoaderSquare
            isInsideButton={isInsideButton}
            isActive={isActive}
            size={size}
           backgroundColor={isGradient ? "#8F49FE" : undefined} 
          />

Reply:
Cool, I'll stick with mapping but use this approach

Reply:
**suggestion**: what do you think about combining the `gradientLoaderColors` with your map to simplify the code?:

```tsx
{["#13A038", "#0092DB", "#8F49FE"].map((color) => (
    <StyledLoaderSquare
      key={color}
      backgroundColor={
        variant === "gradient"
          ? color
          : "var(--colorsActionMajor500)"
      }
      {...loaderSquareProps}
    />
))}
```